applications/feedback_sentiment_analyzer/SPECIFICATION.md

```diff
@@ -0,0 +1,224 @@
+  * [Managing items in DynamoDB](#managing-items-in-dynamodb)
+  * [Other FSA material](#other-fsa-material)
+
+---
+
+## Inputs
+This application receives three inputs from the frontend.
+* authenticate user (uses `Cognito`)
+* load page (uses `DynamoDB:GetItems`)
+* upload image to S3 (`S3:PutObject`)
+
+## Outputs
+This application produces two outputs from the backend:
+* put new item in the database (uses `DynamoDB:PutItem`)
+* put new synthesized audio to S3 (uses `S3:PutObject`)
+
+## Diagram
+
+![img_1.png](img_1.png)
+
+---
+
+## Input routing
+The application inputs diagrammed above are routed through API Gateway.
+
+| action       | route                    | method | request target                                              | response model                                           |
+|--------------|--------------------------|--------|-------------------------------------------------------------|----------------------------------------------------------|
+| Load page    | `/feedback`                | GET    | [GetFeedback Lambda function](#GetFeedback-lambda-function) | `[   {     "text": String,     "audioUrl": String   } ]` |
+| Upload image | `/audio/{bucket}/{object}` | PUT    | [State Machine via s3:PutObject API]()                      | void                                                     |
+
+---
+
+## Step Function configuration
+When an image is uploaded through `Upload image` input route, a Step Function state machine is triggered.
+
+See [Event processing](#event-processing)
+This state machine orchestrates a series of AWS Lambda functions.
+
+### State machine Lambda functions
+
+| trigger                      |        function       | action                                   |
+|------------------------------|-----------------------|------------------------------------------|
+| S3:CreateObject event.       | [ExtractText](#ExtractText)      | Extracts text from   S3 object.          |
+| `ExtractText` complete.      | [AnalyzeSentiment](#AnalyzeSentiment) | Detect positive or   negative sentiment. |
+| `AnalyzeSentiment` complete. | [TranslateText](#TranslateText)    | Translate text to   French.              |
+| `TranslateText` complete.    | [SynthesizeAudio](#SynthesizeAudio)  | Synthesize   human-like audio from text. |
+
+#### Diagram
+These functions appear by name in the multi-state ]workflow depicted below. 
+
+Also depicted is where the workflow forks at `ContinueIfPositive` based whether sentiment analyzed is `POSITIVE` or `NEGATIVE`. 
+
+When ran successfully, the state machine  adds a new item to a DynamDB table containing data gathered during the workflow.
+
+![FSA - l'hotellerie.png](..%2F..%2F..%2F..%2F..%2FDownloads%2FFSA%20-%20l%27hotellerie.png)
+
+Logging appears in the AWS Console under Step Functions and (for Lambda functions only) CloudWatch.
+
+## Function inputs and outputs
+See below for the required inputs and outputs of each Lambda function.
+
+### ExtractText
+
```

Comment:
Each function needs a description of it's behavior. For example:
> ExtractText is the entry point in a series of AWS Step Functions. It's triggered when an image file is uploaded to the Amazon S3 `media-bucket` created when you run `cdk deploy` in [the README](./README.md). The function uses the inputs described below to run [DetectDocumentText](https://docs.aws.amazon.com/textract/latest/dg/API_DetectDocumentText.html) with Amazon Textract. The extracted words are joined together as one string and returned.

Reply:
Start w/ bullet points that are super concise, then link to individual sections.

At DynamoDB step, link down to Dynamo section